README.md

```diff
@@ -743,6 +758,72 @@ minconns: 2
 Note: This functionality should be considered final. If you need to manipulate keys, combine values, or perform mutations, that should be done _outside_ of Consul. In order to keep the API scope limited, we likely will not accept Pull Requests that focus on customizing the `toYAML` functionality.
 
 
+Plugins
+-------
+### Authoring Plugins
+For some use cases, it may be necessary to write a plugin that offloads work to another system. This is especially useful for things that may not fit in the "standard library" of Consul Template, but still need to be shared across multiple instances.
+
+Consul Template plugins must have the following API:
+
+```shell
+$ NAME [INPUT...]
+```
+
+- `NAME` - the name of the plugin - this is also the name of the binary
+- `INPUT` - input from the template - this wil always be JSON if provided
+
+#### Important Notes
+
+- Plugin output must be returned as a JSON string (even if it is empty)
+- Only stdout will be parsed for output - be sure to log all errors, debugging
+  messages onto stderr to avoid errors when Consul Template tries to parse the
+  JSON
+- Always `exit 0` or Consul Template will assume the plugin failed to execute
+
```

Comment:
Maybe we should add a note somewhere in here about security implications, something like:
```
- Plugins execute user-provided scripts and pass in potentially sensitive data from Consul.
  Neither is validated or protected by consul-template, so all necessary precautions and
  considerations should be made.
```